tools/fluent/basic_syntax.md

```diff
@@ -0,0 +1,131 @@
+
+```PROPERTIES
+welcome-msg = Welcome { $user }
+```
+
+`$user` is the name of the argument, and should never be translated. The fragment included between curly braces is called a **placeable**, and can be moved within the text. For example, the string above would be translated in Italian as follows, leaving the placeable unchanged:
+
+```PROPERTIES
+welcome-msg = Benvenuto { $user }
+```
+
+Messages can also reference other messages. In this case, the placeable won’t include the `$` sign, simply the message identifier:
+
+```PROPERTIES
+menu-settings = Settings
+help-menu-settings = Click “{ menu-settings }” to save the file.
+```
+
+There is a special category of messages, called **private messages**:
+* Private messages have identifiers starting with a dash, e.g. `-brand-short-name`.
+* They can’t be called directly in the source code, but only referenced within other messages.
+
+For example:
+
+```PROPERTIES
+-brand-short-name = Firefox
+close-msg = Close { -brand-short-name }
+```
+
+## Attributes
+
+For a single HTML object there might be multiple messages to localize: for example, a button could have a `label` and an `accesskey`, a link could have a text but also a `tooltip`. Such objects can be localized defining multiple **attributes** within a single message. For example, in the case of a button:
+
+```PROPERTIES
+login-button =
+    .label = Login
+    .accesskey = L
+```
+
+The message `login-button` doesn’t have a *value*, but has 2 *attributes* defined: `label` and `accesskey`. There are a few things to highlight in this FTL fragment:
+* Attribute definitions must be indented and start with a period.
+* There is an equal sign following the message identifier, even if there is no value (i.e. the value is *Null*).
+
+```PROPERTIES
+instructions-link = Log out
+    .tooltip = Disconnect from this account
+```
+
+In this case, `instructions-link` has a *value* (`Log out`) and a `tooltip` *attribute* (`Disconnect from this account`).
+
+It’s important to note that, in FTL files, **indendation** is part of the syntax; indenting elements incorrectly will lead to parsing errors.
+
+## Selectors and plurals
+
+With the select expression, a single message can provide several alternatives. The selected value will depend on the value of an external variable or function:
+
+```PROPERTIES
+welcome-pet = { $petFamily ->
+        [canine] Hello dog
+        [feline] Hello cat
+        *[other] Hello pet
+    }
+```
+
+Notice that both the variants and the closing curly brace are indented. The same string can also be written as:
+
+```PROPERTIES
+welcome-pet =
+    { $petFamily ->
+        [canine] Hello dog
+        [feline] Hello cat
+        *[other] Hello pet
+    }
+```
+
+The message displayed by the UI will change based on the value of the external variable `$petFamily`. Notice that one of the variants starts with a `*`: that indicates the **default option**, and it must always be defined in a selector.
```

Comment:
I'd remove the whole pet thing, 'cause it's a counter-example to what I think we should do in ftl.

We might add that back with https://github.com/projectfluent/fluent/issues/80 in a better way.

Reply:
Can you clarify? I think we need an example for selectors before introducing plurals. 

If it's about the example in itself: I'm not superhappy with the current example either, so I'd be happy to swap it with something different.

Reply:
On second thoughts, I guess I could use the German example from that PR?

Reply:
> Can you clarify? I think we need an example for selectors before introducing plurals.

Plurals will be by far the most common use-case for selectors and I think it makes sense to start with them. That's why the example in http://projectfluent.org/fluent/guide/selectors.html uses a number for the selector.

Reply:
The risk of starting with plurals is that it might not be clear that you can have arbitrary, even localized, key names

Reply:
I also agree with @Pike that the pet example will be better solved with https://github.com/projectfluent/fluent/issues/80. As per https://github.com/projectfluent/fluent/wiki/Good-Practices-for-Developers, given the current Fluent spec, I'd rather see three messages for this than a select expression.